client.go

```diff
@@ -122,9 +121,7 @@ func (c *client) Signature(reqSign *sign.RequestSignature) (string, error) {
 func (c *client) Do(ctx context.Context, method, url string, req ...interface{}) *Result {
 	// 1. serialize the request
 	var reqBuffer []byte
-	// now we only use method condition to solve this.
-	if len(req) > 0 && method != http.MethodGet &&
-		req[0] != nil && !reflect.ValueOf(req[0]).IsNil() {
```

Comment:
please use `req[0] != nil && !reflect.ValueOf(req[0]).IsNil()` instread of `req[0] != nil`.    beasure of The request may be `(*PayRequest)(nil)`

Reply:
```
var req = (*PayRequest)(nil)
print(req == nil)   // true
```